rero_ils/modules/stats/cli.py

```diff
@@ -83,20 +83,28 @@ def collect(type):
 @stats.command('collect_year')
 @click.argument('year', type=int)
 @click.argument('timespan', default='yearly')
+@click.option('--n', default=12)
 @click.option('-f', '--force', is_flag=True, default=False)
 @with_appcontext
-def collect_year(year, timespan, force):
+def collect_year(year, timespan, n, force):
     """Extract the stats librarian for one year and store them in db.
 
     :param year: year of statistics
+    :param n: month up to which the statistics are calculated
     :param timespan: time interval, can be 'montly' or 'yearly'
     :param force: force update of stat.
     """
     stat_pid = None
     type = 'librarian'
     if year:
         if timespan == 'montly':
-            for month in range(1, 13):
+            if n in range(1, 13):
+                n += 1
+            else:
+                click.secho(f'ERROR: not a valid month')
+                return
```

Comment:
You can use a `click.Abort` here: https://click.palletsprojects.com/en/7.x/api/#click.Abort